rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -248,7 +248,7 @@ func (s *RollupSyncService) parseAndUpdateRollupEventLogs(logs []types.Log, endB
 					return fmt.Errorf("failed to get local node info, batch index: %v, err: %w", index, err)
 				}
 
-				endBlock, finalizedBatchMeta, err := validateBatch(event, parentBatchMeta, chunks, s.bc.Config(), s.stack)
+				endBlock, finalizedBatchMeta, err := validateBatch(index, event, parentBatchMeta, chunks, s.bc.Config(), s.stack)
```

Comment:
Comment about line 256 below:

It's safer to collect all batch metas, then write all into db after the for loop (or use a batch writer and flush after the for loop). Otherwise we might write some incorrect batches, before the last batch fails verification.

Reply:
switched to using a batch writer for updating the metadata of batches in a bundle.